src/client/components/app/Transactions/EnableSpigotTx.tsx

```diff
@@ -200,32 +203,71 @@ export const EnableSpigotTx: FC<EnableSpigotTxProps> = (props) => {
     );
   }
 
+  const renderRevFunc = () => {
```

Comment:
can u templatize the rev/claim data into an object and do a mapping so we dont have redundant code and its more succinct.

```
const revFuncDisplayConfigs = [{ header: t(), options: [], type: claimFuncType, onChange: onClaim}, { ... }]
const renderFuncSelectors = () => revFuncDisplayConfigs.map(config =>  revContractABI ? (< /> ) : ( < />) )
...
{renderFuncSelectors()}
```
make sense?

Programming Reason:
Right now its kinda imperative. Its wrapped in functions but the functions have repetitive logical branches and code even though they work on the exact same data structures. Its better to define data struct and how process it then just pass data in. 

Functional Reason:
I want it setup this way because i want to add custom multicall tx support for multisigs so sophisticated users and mainly the arbiter can do more complex/automated processes inside the app (e.g. call `liquidate` then trade on 0x and call `depositAndRepay` all at once or `claimRevenue` on Spigot and then `claimAndRepay` on line)

Reply:
done